tests/devices/test_null_qubit.py

```diff
@@ -110,6 +110,77 @@ def test_tracking():
     }
 
 
+def test_assume_no_broadcasting():
+    """Test that if no batch sizes are added if ``assume_no_broadcasting`` is True."""
+
+    dev = qml.device("null.qubit", assume_no_broadcasting=True)
+
+    tape = qml.tape.QuantumScript([qml.RX((1.2, 2.3), 0)], [qml.expval(qml.Z(0))])
+    results = dev.execute(tape)
+    assert qml.math.allclose(results, np.array(0.0))
```

Comment:
How does this test no broadcasting? 🤔

Reply:
the tape has a broadcast dimension.  So if it actually checked, the output would have an extra dimension.